crates/uv-resolver/src/universal_marker.rs

```diff
@@ -143,6 +141,43 @@ impl UniversalMarker {
         self.pep508 = self.marker.without_extras();
     }
 
+    /// If all inference sets reduce to the same marker, simplify the marker using that knowledge.
```

Comment:
@BurntSushi Is it correct to assume that if we have inferences, we must be in one of them, i.e. there can't be a missing "empty" inference set?